### PR TITLE
fix: Support property scoping in the UI.

### PR DESCRIPTION
**Pull request by pleacu**
Fixes: #2209

### CODE REVIEW

File: lib/model/src/main/java/io/atlasmap/v2/Property.java [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
     protected FieldType fieldType;
 
+    protected String scope;
```
Comment (igarashitm) [inline lib/model/src/main/java/io/atlasmap/v2/Property.java:15]: Was it easier to put `scope` here as well? This `Property` is used to define a fallback value apart from `PropertyField`, and `scope` only makes sense on `PropertyField`.

File: ui/packages/atlasmap/src/Atlasmap/utils/document.ts [hunk 2]
```diff
@@ -329,3 +329,30 @@ export function enableCustomClass(
...
+    },
+    {
+      label: "Camel Exchange Property",
+      value: "camelExchangeProperty",
+    },
+  ];
+  for (let i = 0; i < cfg.sourceDocs.length; i++) {
+    scopeOptions.push({
+      value: cfg.sourceDocs[i].id,
+      label: cfg.sourceDocs[i].id,
```
Comment (igarashitm) [inline ui/packages/atlasmap/src/Atlasmap/utils/document.ts:354]: Maybe `cfg.sourceDocs[i].name` for label?

Comment (igarashitm) [reply ui/packages/atlasmap/src/Atlasmap/utils/document.ts:354]: Also, wonder if it's possible to show `cfg.sourceDocs[i].description` as a tooltip?

File: lib/model/src/main/java/io/atlasmap/v2/Property.java (hunk 1 above)
Comment (pleacu) [reply lib/model/src/main/java/io/atlasmap/v2/Property.java:15]: Needed to set the scope argument when editing an existing property.

**Pull request merged**